scripts/release.py

```
@@ -0,0 +1,362 @@
+    with open(path, "w", encoding="utf-8") as f:
+        f.write(text)
+
+
+def parse_version(text):
+    parts = text.split(".")
+    if len(parts) != 3:
+        raise ReleaseError("Malformed version '%s', expected M.m.P" % text)
+    try:
+        return tuple(int(p) for p in parts)
+    except ValueError:
+        raise ReleaseError("Malformed version '%s', expected M.m.P" % text)
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)